libs/dh/core/shell/src/lib/dh-core-login.component.ts

```diff
@@ -117,31 +119,42 @@ import { DhFeatureFlagDirective } from '@energinet-datahub/dh/shared/feature-fla
   `,
 })
 export class DhCoreLoginComponent implements AfterViewInit {
-  private _config = inject(dhB2CEnvironmentToken);
+  private config = inject(dhB2CEnvironmentToken);
+  private appInsights = inject(DhApplicationInsights);
 
   progressBarValue = signal(0);
   showProgressBar = signal(false);
 
-  @ViewChild(DhMitIDButtonComponent) mitIdButton!: DhMitIDButtonComponent;
+  mitIdButton = viewChild.required(DhMitIDButtonComponent);
 
   ngAfterViewInit(): void {
     const mitIdRelogin = Boolean(localStorage.getItem('mitIdRelogin'));
+
     if (mitIdRelogin) {
       this.showProgressBar.set(true);
+
       setInterval(() => {
         this.progressBarValue.set(this.progressBarValue() + 1);
       }, 200);
+
       setTimeout(() => {
         localStorage.removeItem('mitIdRelogin');
-        this.mitIdButton.redirectToMitIdSignup();
-        this.showProgressBar.set(false);
-      }, 20000);
```

Comment:
Issue: Before it redirected after 20 sec, now it does it after 2 sec?

Reply:
Not really. Now it redirects after 22 sec - 20 sec for the outer `setTimeout` + 2 sec for the inner where the redirect is triggered.

Reply:
You are right :-)